roles/openshift_hosted/filter_plugins/filters.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+'''
+Custom filters for use in openshift_hosted
+'''
+
+
+class FilterModule(object):
+    ''' Custom ansible filters for use by openshift_hosted role'''
+
+    @staticmethod
+    def get_router_replicas(replicas=None, router_nodes=None):
+        ''' This function will return the number of replicas
+            based on the results from the defined
+            openshift.hosted.router.replicas OR
+            the query from oc_obj on openshift nodes with a selector OR
+            default to 1
+
+        '''
+        # We always use what they've specified if they've specified a value
+        if replicas is not None:
+            return replicas
+
+        if (isinstance(router_nodes, dict) and
+                'results' in router_nodes and
+                'results' in router_nodes['results'] and
+                'items' in router_nodes['results']['results']):
+
+            return len(router_nodes['results']['results'][0]['items'])
```

Comment:
Would it be better to not be so tightly coupled to the module result implementation here? It looks like if the module result format changes, then we could return 1 instead of the number of nodes here.